logstash-core/spec/logstash/instrument/metric_store_spec.rb

```diff
@@ -0,0 +1,103 @@
+# encoding: utf-8
+require "logstash/instrument/metric_store"
+require "logstash/instrument/metric_type/base"
+
+describe LogStash::Instrument::MetricStore do
+  let(:namespaces) { [ :root, :pipelines, :pipeline_01 ] } 
+  let(:key) { :events_in }
+  let(:counter) { LogStash::Instrument::MetricType::Counter.new(namespaces, key) }
+
+  context "when the metric object doesn't exist" do
+    it "store the object" do
+      expect(subject.fetch_or_store(namespaces, key, counter)).to eq(counter)
+    end
+
+    it "support a block as argument" do
+      expect(subject.fetch_or_store(namespaces, key) { counter }).to eq(counter)
+    end
+  end
+
+  context "when the metric object exist in the namespace"  do
+    let(:new_counter) { LogStash::Instrument::MetricType::Counter.new(namespaces, key) }
+
+    before do
+      subject.fetch_or_store(namespaces, key, counter)
+    end
+
+    it "return the object" do
+      expect(subject.fetch_or_store(namespaces, key, new_counter)).to eq(counter)
+    end
+  end
+
+  context "when the namespace end node isn't a map" do
+    let(:conflicting_namespaces) { [:root, :pipelines, :pipeline_01, :events_in] }
+
+    before do
+      subject.fetch_or_store(namespaces, key, counter)
+    end
+
+    it "raise an exception" do
+      expect { subject.fetch_or_store(conflicting_namespaces, :new_key, counter) }.to raise_error(LogStash::Instrument::MetricStore::ConcurrentMapExpectedError)
+    end
+  end
+
+  describe "#get" do
+    let(:metric_events) {
+      [
+        [[:node, :sashimi, :pipelines, :pipeline01, :plugins, :"logstash-output-elasticsearch"], :event_in, :increment],
```

Comment:
I wanted to use a symbol and not a string, since all the other are symbols.

Symbols doesn't allow you to use `-` init unless you use quotes, without quote it think you are actually doing  subtraction.

```
jruby-1.7.23 :002 > :logstash-output-eleasticsearch
NameError: undefined local variable or method `output' for main:Object
```

with quotes:
```
jruby-1.7.23 :003 > :"logstash-output-eleasticsearch"
 => :"logstash-output-eleasticsearch"
```